python/pydantic_core/_pydantic_core.pyi

```diff
@@ -822,54 +822,171 @@ class ValidationError(ValueError):
+
+    Example:
+    ```python
+    from pydantic_core import PydanticCustomError
+
+    def custom_validator(v) -> None:
+        if v <= 10:
+            raise PydanticCustomError('custom_value_error', 'Value must be greater than {value}', {'value': 10})
+        return v
+    ```
+    """
+
     @property
     def context(self) -> dict[str, Any] | None: ...
+    """Values which are required to render the error message, and could hence be useful in passing error data forward.
+
+    For example, if your message template was `"{field_name} must be greater than {value}"`, you could pass
+    `{'field_name': 'age', 'value': 18}` as the context.
+    """
+
     @property
     def type(self) -> str: ...
+    """The error type associated with the error. For consistency with Pydantic, this is typically a snake_case string."""
+
     @property
     def message_template(self) -> str: ...
+    """The message template associated with the error. This is a string that can be formatted with context variables in {curly_braces}.
+
+    For example, if your message template was `"{field_name} must be greater than {value}"`, you could pass
+    `{'field_name': 'age', 'value': 18}` as the context.
+    """
+
     def message(self) -> str: ...
+    """The formatted message associated with the error. This presents as the message template with context variables appropriately injected."""
 
 @final
 class PydanticKnownError(ValueError):
+    """A helper class for raising exceptions that mimic Pydantic's built-in exceptions, with more flexibility in regards to context.
+
+    Unlike `PydanticCustomError`, the `error_type` argument must be a known [`ErrorType`][pydantic_core.ErrorType]."""
+
     def __new__(cls, error_type: ErrorType, context: dict[str, Any] | None = None) -> Self: ...
+    """Create a new `PydanticKnownError`.
+
+    Arguments:
+        error_type: The error type, one of [`ErrorType`][pydantic_core.ErrorType].
+        context: The data to inject into the message template.
+    """
+
     @property
     def context(self) -> dict[str, Any] | None: ...
+    """Values which are required to render the error message, and could hence be useful in passing error data forward."""
+
     @property
     def type(self) -> ErrorType: ...
+    """The type of the error, one of [`ErrorType`][pydantic_core.ErrorType]."""
+
     @property
     def message_template(self) -> str: ...
+    """The message template associated with the provided error type. This is a string that can be formatted with context variables in {curly_braces}."""
+
     def message(self) -> str: ...
+    """The formatted message associated with the error. This presents as the message template with context variables appropriately injected."""
 
 @final
 class PydanticOmit(Exception):
+    """An exception to signal that a field should be omitted from generated JSON schema.
+
+    This exception can be raised from custom JSON schema generation functions to indicate that
```

Comment:
This seems like a very specific use case, better to give a more general case.

Reply:
Yeah, we don't actually support super general use cases right now (like `PydanticOmit` in a custom serializer), so I've added a note and an example here.